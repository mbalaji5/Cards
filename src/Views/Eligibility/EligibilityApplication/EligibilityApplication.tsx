import React, { useState } from "react";
import { useFormik } from "formik";
import styled from "styled-components";

import FormInput from "../../../DesignSystem/Form/FormInput";
import SubmitButton from "../../../DesignSystem/Form/SubmitButton";
import Title from "../../../DesignSystem/Title";
import Error from "../../Error/Error";

import { IEligibilityRequest } from "../../../DataModel/eligibility/eligibility";

const FormWrapper = styled.div`
  flex: 1 1 auto;
  width: 100%;
`;

interface FormValues {
  name: string;
  email: string;
  address: string;
}
interface propType {
  eligibilityCheckHandler: (reqData: IEligibilityRequest) => {};
}

const EligibilityApplication = (props: propType) => {
  const [errorMessage, setErrorMessage] = useState("");
  const { handleChange, handleSubmit, values } = useFormik<FormValues>({
    initialValues: {
      name: "",
      email: "",
      address: "",
    },
    onSubmit: (values) => {
      if (!values.name || !values.email || !values.address) {
        setErrorMessage("Please fill all the fields!");
      } else {
        props.eligibilityCheckHandler(values);
        setErrorMessage("");
      }
    },
  });
  return (
    <FormWrapper>
      <Title>Cards</Title>
      <form onSubmit={handleSubmit}>
        <FormInput
          type="text"
          name="name"
          id="name"
          data-testid="name"
          onChange={handleChange}
          value={values.name}
          placeholder="Name"
        />
        <FormInput
          type="email"
          name="email"
          id="email"
          data-testid="email"
          onChange={handleChange}
          value={values.email}
          placeholder="Email"
        />
        <FormInput
          type="text"
          name="address"
          id="address"
          data-testid="address"
          onChange={handleChange}
          value={values.address}
          placeholder="Address"
        />
        <SubmitButton id="submit" text="Submit" />
      </form>
      {errorMessage && <Error errorMessage={errorMessage} />}
    </FormWrapper>
  );
};

EligibilityApplication.defaultProps = { eligibilityCheckHandler: () => {} };
export default EligibilityApplication;
