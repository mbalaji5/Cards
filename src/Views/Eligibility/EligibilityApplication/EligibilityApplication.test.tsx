import { fireEvent, render, act } from "@testing-library/react";
import React from "react";
import EligibilityApplication from "./EligibilityApplication";

describe("Should Validate EligibilityApplication Component", () => {
  it("Should validate eligibilityResult With Valid C1 and C2", () => {
    const eligibilityResult = {
      cardType: ["C1", "C2"],
      eligibility: {
        C1: { isEligible: true },
        C2: { isEligible: true },
      },
    };

    const { container, baseElement } = render(
      <EligibilityApplication eligibilityCheckHandler={() => jest.fn()} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
