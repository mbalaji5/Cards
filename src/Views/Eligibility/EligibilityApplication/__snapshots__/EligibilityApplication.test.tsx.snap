// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Validate EligibilityApplication Component Should validate eligibilityResult With Valid C1 and C2 1`] = `
<body>
  <div>
    <div
      class="sc-jSgvzq ekxYJa"
    >
      <h1
        class="sc-eCstlR jAIppu"
      >
        Cards
      </h1>
      <form>
        <div
          class="sc-bdfBQB LQheS"
        >
          <input
            class="sc-gsTEea fptcsq"
            data-testid="name"
            id="name"
            name="name"
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-bdfBQB LQheS"
        >
          <input
            class="sc-gsTEea fptcsq"
            data-testid="email"
            id="email"
            name="email"
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
        <div
          class="sc-bdfBQB LQheS"
        >
          <input
            class="sc-gsTEea fptcsq"
            data-testid="address"
            id="address"
            name="address"
            placeholder="Address"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-dlfnuX jARWZH"
        >
          <button
            class="sc-hKgJUU TlIWR"
            data-testid="submit"
            id="submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
      
    </div>
  </div>
</body>
`;
